### PR TITLE
Provision for showing only filter container and hiding the "search tool" button on list view.

### PR DESCRIPTION
**Pull request by amit4106udale**
### Is your feature request related to a problem? Please describe.
Yes, Currently there is no such provision to show filters container by hiding the "search tools" button. Most of the times in frontend, user don't want an extra click to open the filters. So, we need a facility to show only the filters container of the search tool.

users - joomlafilters - administration

### Describe the solution you'd like
Need to add a config in custom options for showing the filter container and hide the "search tools button". 

### Additional context
It will reduce the number of clicks for better user experience. 
Sometimes there are only one or to filter fileds in filter container so its is not needed show the  "search tools" button. Instead of that, we can show the filter container directly.

**For example :-**
We need to pass one extra param at the time of rendering respective search toolbar, on the basis of this it will hide the  "search tools" button and directly show the filter container.

### **Testing Instructions**
While rendering search tools use showOnlyFilters parameter and set it as 1.

  **JLayoutHelper::render('joomla.searchtools.default', array('view' => $this, 'options' => array('showOnlyFilters' => 1)));**

### CODE REVIEW

**Comment by mbabker** [comment]
There are already other options to configure this behavior if I've read the layout file correctly.  So what is adding this option actually accomplishing that can't already be done?

**Comment by vaibhavsTekdi** [comment]
@mbabker - We don't have such configuration,  we have filterHidden -> which always keeps the "search tool" open, and filterButton-> which hides button as well as the container.

As you can see below screenshot, when we apply filtersHidden config -> it gives such display,
where "search tools" button is visble, which confuses the users.

screenshot-localhost-2019 01 11-18-29-02

**Comment by amit4106udale** [comment]
@mbabker -> as you can check the old code "`<?php if ($data['options']['filterButton']) : ?>
	<div class="js-stools-container-filters hidden-phone clearfix<?php echo $filtersActiveClass; ?>">
		<?php echo $this->sublayout('filters', $data); ?>
	</div> ` "

 filters are dependent  on "search tool" button.

If I use the filters layout directly to show the filters, the clear button won't work for the filters.

**Comment by ReLater** [comment]
I have tested this item :white_check_mark: successfully on b1c47d968fd592be018681e88d8ebe31b2dcb8b6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23508">issues.joomla.org/tracker/joomla-cms/23508</a>.</sub>

**Comment by uthorat** [comment]
I have tested this item :white_check_mark: successfully on b1c47d968fd592be018681e88d8ebe31b2dcb8b6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23508">issues.joomla.org/tracker/joomla-cms/23508</a>.</sub>

**Comment by pranotiTechjoomla** [comment]
I have tested this item :white_check_mark: successfully on b1c47d968fd592be018681e88d8ebe31b2dcb8b6<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23508">issues.joomla.org/tracker/joomla-cms/23508</a>.</sub>

**Comment by pranotiTechjoomla** [comment]
RTC

**Comment by infograf768** [comment]
maybe I did not understand the right code to use.
I tested com_associations in `/administrator/components/com_associations/views/associations/tmpl/default.php`

I modified the call there to 
`<?php echo JLayoutHelper::render('joomla.searchtools.default', array('view' => $this, options => array('showOnlyFilters' => 1))); ?>` to test.

I get a Notice:
`Notice: Use of undefined constant options - assumed 'options' in .../administrator/components/com_associations/views/associations/tmpl/default.php on line 63`

**Comment by ReLater** [comment]
Just a typo in the code. Use quoted `'options' => ...` not `options => ...`!

**Comment by infograf768** [comment]
I just copied what was in the original PO...
Will modify it there now.

File: media/jui/js/jquery.searchtools.js [hunk 1]
```diff
@@ -55,6 +55,9 @@
 		this.filterContainer = $(this.options.formSelector + ' ' + this.options.filterContainerSelector);
 		this.filtersHidden   = this.options.filtersHidden;
 
+		// Only Filters Without 'Search Tools' Button
```
Comment (Quy) [inline media/jui/js/jquery.searchtools.js]:
```suggestion
		// Only filters without 'Search Tools' button
```

File: media/jui/js/jquery.searchtools.js [hunk 2]
```diff
@@ -55,6 +55,9 @@
...
+		this.showOnlyFilters   = this.options.showOnlyFilters;
```
Comment (Quy) [inline media/jui/js/jquery.searchtools.js]:
```suggestion
		this.showOnlyFilters = this.options.showOnlyFilters;
```

File: media/jui/js/jquery.searchtools.js (hunk 1 above)
Comment (amit4106udale) [reply media/jui/js/jquery.searchtools.js]: @Quy - done

File: media/jui/js/jquery.searchtools.js (hunk 2 above)
Comment (amit4106udale) [reply media/jui/js/jquery.searchtools.js]: @Quy - done

File: media/jui/js/jquery.searchtools.js [hunk 3]
```diff
@@ -106,10 +109,15 @@
 			// Get values
 			this.searchString = this.searchField.val();
 
-			if (this.filtersHidden) {
-				this.hideFilters();
-			} else {
+			// Check if only filters is need without 'search tool' button
+			if (this.showOnlyFilters) {
```
Comment (Quy) [inline media/jui/js/jquery.searchtools.js:113]: This can be simplified to be one level (pseudo code):
```
if
else if
else
```

Comment (amit4106udale) [reply media/jui/js/jquery.searchtools.js:113]: @Quy -Done

Comment (Quy) [reply media/jui/js/jquery.searchtools.js:113]: Please update the minified js file. Thanks.

Comment (amit4106udale) [reply media/jui/js/jquery.searchtools.js:113]: @Quy - Already minified. Again minified but it is not showing any changes in the file.

**Comment by ReLater** [comment]
Shouldn't this PR get a RTC

OR

shouldn't it be closed because "No new features allowed in Joomla 3!"

**Comment by ghost** [comment]
calling @HLeithner as Release Lead J3 for Comment above.

**Comment by HLeithner** [comment]
Sorry to say that's a new feature and 3.x has a feature freeze plz get involved into the J4 development.
The backend template team is rewriting many parts of the template code so maybe you check the current j4 backend template code base at https://github.com/joomla/backend-template

**Pull request closed without merging**